### PR TITLE
Code Insights: Add insight export data UI

### PR DESCRIPTION
**Pull request by vovakulikov**
Closes https://github.com/sourcegraph/sourcegraph/issues/46493

This PR adds export data UI, (link in the three dot context menu and button on the standalone insight page) This PR should be merged after the [BE link implementation](https://github.com/sourcegraph/sourcegraph/pull/46662) is merged. 

## Test plan
- Open the three dot context menu and export the data 
- It should be available for all insights except lang stats
- Go to the standalone insight page (click insight card title) 
- Try to export data with header export data button

## App preview:

- [Web](https://sg-web-vk-add-export-ui.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.01% (+1.27 kb) | 0.01% (+1.27 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/vk/add-export-ui/compare-report.html;tab=live_object) for a full comparison between the commits ee01b0f and 02c985e or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/enterprise/insights/pages/insights/insight/components/actions/CodeInsightIndependentPageActions.tsx [hunk 1]
```diff
@@ -46,6 +46,12 @@ export const CodeInsightIndependentPageActions: FunctionComponent<Props> = props
 
     return (
         <div className={styles.container}>
+            {!isLangStatsInsight(insight) && (
+                <Button as={Link} to={`/api/insights/export/${insight.id}`} variant="secondary">
```
Comment (leonore) [inline client/web/src/enterprise/insights/pages/insights/insight/components/actions/CodeInsightIndependentPageActions.tsx:50]: `insight.id` corresponds to the id we put in the url right?

File: client/web/src/enterprise/insights/pages/insights/insight/components/actions/CodeInsightIndependentPageActions.tsx [hunk 2]
```diff
@@ -46,6 +46,12 @@ export const CodeInsightIndependentPageActions: FunctionComponent<Props> = props
...
+                    Export CVS data
```
Comment (leonore) [inline client/web/src/enterprise/insights/pages/insights/insight/components/actions/CodeInsightIndependentPageActions.tsx:51]:
```suggestion
                    Export data as CSV
```
I think that's a more standard way of saying it

**Review by leonore — COMMENTED** [review]
I'll give an approval once I've tested this with the backend landed but thank you for working on this so quickly!!

File: client/web/src/enterprise/insights/pages/insights/insight/components/actions/CodeInsightIndependentPageActions.tsx (hunk 1 above)
Comment (vovakulikov) [reply client/web/src/enterprise/insights/pages/insights/insight/components/actions/CodeInsightIndependentPageActions.tsx:50]: Yes

File: client/web/src/enterprise/insights/components/modals/ExportInsightDataModal.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            <Text>This will only include data that you are permitted to see.</Text>
+            <div className="d-flex justify-content-end mt-5">
+                <Button
+                    autoFocus={true}
+                    as={Link}
+                    to={`/api/insights/export/${insightId}`}
+                    variant="primary"
+                    onClick={onConfirm}
+                >
+                    Export CVS data
```
Comment (leonore) [inline client/web/src/enterprise/insights/components/modals/ExportInsightDataModal.tsx:32]:
```suggestion
                    Export data as CSV
```

File: client/web/src/enterprise/insights/components/modals/ExportInsightDataModal.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+            <Text className="mt-4 mb-2">
+                This will create a CVS archive of all data for this Code Insight, including data that has been archived.
+            </Text>
+            <Text>This will only include data that you are permitted to see.</Text>
+            <div className="d-flex justify-content-end mt-5">
+                <Button
+                    autoFocus={true}
+                    as={Link}
+                    to={`/api/insights/export/${insightId}`}
```
Comment (leonore) [inline client/web/src/enterprise/insights/components/modals/ExportInsightDataModal.tsx:28]:
```suggestion
                    to={`/.api/insights/export/${insightId}`}
```

File: client/web/src/enterprise/insights/pages/insights/insight/components/actions/CodeInsightIndependentPageActions.tsx (hunk 1 above)
Comment (leonore) [inline client/web/src/enterprise/insights/pages/insights/insight/components/actions/CodeInsightIndependentPageActions.tsx:50]:
```suggestion
                <Button as={Link} to={`/.api/insights/export/${insight.id}`} variant="secondary">
```

**Review by leonore — COMMENTED** [review]
landed the BE part so tested it a little. see commit suggestions for functional fixes
1- do you know why from the webapp the url routes to the repository page rather than the internal API endpoint? 
i.e. https://sourcegraph.test:3443/.api/insights/export/aW5zaWdodF92aWV3OiIyR2Q0Y1RwbzdLb2d5aGkzMUNuQTQ3amRFZG4i -> `repo:^\.api/insights/export/aW5zaWdodF92aWV3OiIyR2Q0Y1RwbzdLb2d5aGkzMUNuQTQ3amRFZG4i$`  search

2- is there a reason we don't allow the "Export data as CSV" option from the 3-dot menu?
3- I couldn't see the information pop up from the single insight page, it just linked straight to the API link.

**Comment by vovakulikov** [comment]
Thanks for checking @leonore 
1. `do you know why from the webapp the url routes to the repository page rather than the internal API endpoint?` I think there was a problem that I used link without download attribute and client routing use it as absolute path to redirect you to the repository page. It should be fixed now. 

2. `is there a reason we don't allow the "Export data as CSV" option from the 3-dot menu?` Again my mistake, I fixed the rendering check

3. `I couldn't see the information pop up from the single insight page, it just linked straight to the API link.` This one was done intentionally actually :) I think on the dashboard page it's okay to have intermediate step but on the single insight page instead of big loud modal we could use humble and quite tooltip that says the same thing as modal but doesn't introduce additional click. 

Screenshot 2023-01-23 at 19 34 40

**Review by chwarwick — APPROVED** [review]

**Pull request merged**